### PR TITLE
Fix blocked domain appears from account feed

### PR DESCRIPTION
**Pull request by tribela**
AccountStatusesFilter wasn't filtering domain blocks

### CODE REVIEW

File: app/models/account_statuses_filter.rb [hunk 1]
```diff
@@ -55,7 +55,14 @@ def filtered_scope
   end
 
   def filtered_reblogs_scope
-    Status.left_outer_joins(:reblog).where(reblog_of_id: nil).or(Status.where.not(reblogs_statuses: { account_id: current_account.excluded_from_timeline_account_ids }))
+    Status
+      .left_outer_joins(:reblog).where(reblog_of_id: nil)
+      .joins('LEFT OUTER JOIN accounts reblogs_accounts ON reblogs_accounts.id = reblogs_statuses.account_id')
```
Comment (ClearlyClaire) [inline app/models/account_statuses_filter.rb:60]: I'm a bit confused by the position of the `where` here, wouldn't it be more understandable like this?
```suggestion
      .left_outer_joins(:reblog)
      .joins('LEFT OUTER JOIN accounts reblogs_accounts ON reblogs_accounts.id = reblogs_statuses.account_id')
      .where(reblog_of_id: nil)
```

An even more concise and idiomatic way may be:
```suggestion
    .left_outer_joins(reblog: :account)
    .where(reblog_of_id: nil)
```

Though that would rename `reblogs_accounts` to `accounts` and that may end up more confusing and more brittle.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Looks fine except for the inline comment.

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
(The performance implications should be relatively minor, but I'm not sure whether we should merge that for 4.1.0 without evaluating them thoroughly first)

**Comment by ClearlyClaire** [comment]
Oops! sorry we forgot this PR!

I'm going to create a new PR with this changes, as this is going to be easier for us to test. Thank you for your contribution!

**Comment by tribela** [comment]
Closing this by #26823

**Pull request closed without merging**